src/controllers/UserController.ts

```diff
@@ -72,6 +76,23 @@ export class UserController {
 		}
 	}
 
+	public async reportUser(req: Request & { params: { id: string } }, res: AuthenticatedResponse, next: NextFunction): Promise<void> {
+		try {
+			if (!req.params.id) throw new APIError(HttpCode.NotFound, GetUserError.UserNotFound);
+			if (req.params.id === '@me') throw new APIError(HttpCode.NotFound, ReportUserError.ReportInvalid);
+			const user = await this.userService.findOne({ id: req.params.id });
+			if (!user) throw new APIError(HttpCode.NotFound, GetUserError.UserNotFound);
+			const report = await this.userService.reportUser(res.locals.user.id, user, req.body);
```

Comment:
The checks above should really be done in the service instead, as that's where all the logic/validation should happen. The controller is basically here to just call the service and respond to a HTTP request accordingly.